### ISSUE TITLE
Spurious PLE1300 error with 0.0.283

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [astral-sh/ruff#6442]
~~After upgrading v0.0.281 → v0.0.282 (https://github.com/codespell-project/codespell/pull/3011)~~ we see what we think is a spurious PLE1300 error:

* Minimal code snippet:
  ```python
  s = "{0}{1:{width}}".format("a", "1", width=15)
  ```
* The command invoked:
   ```
   $ ruff --select PLE file.py 
   file.py:1:5: PLE1300 Unsupported format character '}'
   Found 1 error.
   $ 
   ```
* The current Ruff settings: no `pyproject.toml` file
* The current Ruff version:
  ```
  $ ruff --version
  ruff 0.0.283
  $ 
  ```

The [Format String Syntax](https://docs.python.org/3/library/string.html#format-string-syntax) documentation states:
> A _format_spec_ field can also include nested replacement fields within it. These nested replacement fields may contain a field name, conversion flag and format specification, but deeper nesting is not allowed.

**Comment by MichaReiser** [comment]
This is probably related to #6171

**Comment by DimitriPapadopoulos** [comment]
Perhaps I don't understand how pre-commit checks work. Out file `.pre-commit-config.yaml`  contains:
```yaml
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.0.282
    hooks:
      - id: ruff
``

**Comment by DimitriPapadopoulos** [comment]
But yes, you're right about it:
```
$ pip install --force-reinstall --user ruff==0.0.281
Collecting ruff==0.0.281
  Downloading ruff-0.0.281-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.6/5.6 MB 11.0 MB/s eta 0:00:00
Installing collected packages: ruff
  Attempting uninstall: ruff
    Found existing installation: ruff 0.0.283
    Uninstalling ruff-0.0.283:
      Successfully uninstalled ruff-0.0.283
Successfully installed ruff-0.0.281

[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: python3 -m pip install --upgrade pip
$ 
$ ruff --select PLE file.py 
$ 
$ pip install --force-reinstall --user ruff==0.0.282
Collecting ruff==0.0.282
  Downloading ruff-0.0.282-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.6/5.6 MB 12.6 MB/s eta 0:00:00
Installing collected packages: ruff
  Attempting uninstall: ruff
    Found existing installation: ruff 0.0.281
    Uninstalling ruff-0.0.281:
      Successfully uninstalled ruff-0.0.281
Successfully installed ruff-0.0.282

[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: python3 -m pip install --upgrade pip
$ 
$ ruff --select PLE file.py 
$ 
$ pip install --force-reinstall --user ruff==0.0.283
Collecting ruff==0.0.283
  Using cached ruff-0.0.283-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.7 MB)
Installing collected packages: ruff
  Attempting uninstall: ruff
    Found existing installation: ruff 0.0.282
    Uninstalling ruff-0.0.282:
      Successfully uninstalled ruff-0.0.282
Successfully installed ruff-0.0.283

[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: python3 -m pip install --upgrade pip
$ 
$ ruff --select PLE file.py 
file.py:1:5: PLE1300 Unsupported format character '}'
Found 1 error.
$ 
```
I keep getting issues with pre-commit, whether in CI jobs or on my own computer.

**Comment by kurtmckee** [comment]
> pre-commit

ruff v0.0.283 is getting picked up by an unpinned ruff dev dependency running in a separate CI job. pre-commit is running v0.0.282 as specified.

**Comment by charliermarsh** [comment]
Thanks! I'll take a look at the format string parser. I wonder if we were silently parsing this incorrectly the whole time, and the rule just brought it to light, or if the changes to the parser from https://github.com/astral-sh/ruff/pull/6171 were erroneous.

**Comment by charliermarsh** [comment]
I think the parser doesn't quite support this, it'll need to be improved. (The rule was new in v0.0.283.)

**Comment by tdulcet** [comment]
This is still broken unfortunately. `test.py`:
```py
import sys

number = 1234.56
print("{0:.{prec}g}".format(number, prec=sys.float_info.dig))
```
Ruff output:
```
$ ruff --select PLE1300 --isolated test.py
test.py:4:7: PLE1300 Unsupported format character 'g'
Found 1 error.
```
Ruff version:
```
$ ruff --version
ruff 0.0.285
```

**Comment by DimitriPapadopoulos** [comment]
@tdulcet Perhaps you should open a new issue, since this one has been automatically closed.

**Comment by DimitriPapadopoulos** [comment]
I have opened #6767 as a follow-up to this ticket.
